servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -293,6 +289,11 @@ public void cleanupUser(long userId) throws IOException {
         LOG.error("Error in cleanup userId {}: cannot delete directory ", userId, dirName);
       }
     }
+
+    mEvictionLock.readLock().lock();
+    mMetaManager.cleanupUser(userId);
```

Comment:
Good catch, its actually a pretty subtle corner case. I've adjusted the code to deal with it, but its not as clean as before.